torchgeo/datasets/utils.py

```diff
@@ -255,6 +258,50 @@ def intersects(self, other: "BoundingBox") -> bool:
         )
 
 
+def disambiguate_timestamp(date_str: str, format: str) -> Tuple[float, float]:
+    """Disambiguate partial timestamps.
+
+    Args:
+        date_str: string representing date and time of a data point
+        format: format codes accepted by :meth:`datetime.datetime.strptime`
+
+    Returns:
+        (mint, maxt) tuple for indexing
+    """
+    mint = datetime.strptime(date_str, format)
+
+    # TODO: This doesn't correctly handle literal `%%` characters in format
+
+    # TODO: This is really tedious, is there a better way to do this?
+
+    if not any([f"%{c}" in format for c in "yYcxG"]):
+        # No temporal info
+        return 0, sys.maxsize
+    elif not any([f"%{c}" in format for c in "bBmjUWcxV"]):
+        # Year resolution
+        maxt = datetime(mint.year + 1, 1, 1) - timedelta(microseconds=1)
+    elif not any([f"%{c}" in format for c in "aAwdjcxV"]):
+        # Month resolution
+        maxt = datetime(mint.year, mint.month + 1, 1) - timedelta(microseconds=1)
```

Comment:
Just realized this has a pretty major bug in it, it crashes if `mint.month` is 12.